src/internal/routing-table-getter/routing-table-getter-factory.js

```diff
@@ -43,14 +45,20 @@ export default class RoutingTableGetterFactory {
    * @param {Connection} connection the connection to use
    * @param {string} database the database name
    * @param {string} routerAddress the URL of the router.
-   * @returns {ProcedureRoutingTableGetter} The routing table getter
+   * @returns {ProcedureRoutingTableGetter | RouteMessageRoutingTableGetter} The routing table getter
    */
   create (connection) {
-    const runner =
-      connection.protocol().version < BOLT_PROTOCOL_V4_0
-        ? new SingleDatabaseRoutingProcedureRunner()
-        : new MultiDatabaseRoutingProcedureRunner(this._initialAddress)
+    if (connection.protocol().version < BOLT_PROTOCOL_V4_3) {
```

Comment:
I know this is old code but isn't the point of having a bolt protocol version inheritance structure to handle things like this?

Reply:
Are you saying about the factory code or all the get routing table implementation?

Reply:
Just questioning the need for a factory that checks protocol versions when there already are a protocol inheritance hierarchy that should know type of routing to use for that specific protocol version.

Reply:
Yes, the question is really pertinent. This code factory was created by me on the previous pr, the code before was doing this check really deep on the `RoutingUtil` class and it was making the code inflexible (see #645). 

I think almost all the code should go to the protocol (at the least until the part which get the raw routing table). I just need to think about how to handle some circular dependencies which could occurs if the code got moved as it is.